cmd/armadactl/cmd/cancel.go

```diff
@@ -58,3 +61,30 @@ func cancelJobSetCmd() *cobra.Command {
 	}
 	return cmd
 }
+
+func cancelExecutorCmd() *cobra.Command {
+	a := armadactl.New()
+	cmd := &cobra.Command{
+		Use:   "executor <executor> <priority_class_1> <priority_class_2> ...",
+		Short: "Cancels jobs on executor.",
+		Long:  `Cancels jobs on executor with provided executor name, priority classes and queues.`,
+		Args:  cobra.MinimumNArgs(2),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return initParams(cmd, a.Params)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			onExecutor := args[0]
+			onPriorityClasses := args[1:]
+
+			matchQueues, err := cmd.Flags().GetStringSlice("match-queues")
+			if err != nil {
+				return fmt.Errorf("error reading queue selection: %s", err)
+			}
+
+			return a.CancelOnExecutor(onExecutor, matchQueues, onPriorityClasses)
+		},
+	}
+
+	cmd.Flags().StringSliceP("match-queues", "q", []string{}, "Cancel jobs on executor matching the specified queue names. If no queues are provided, jobs across all queues will be cancelled.")
+	return cmd
```

Comment:
Suggest
- rename `match-queues` to just `queues`
- for consistency, if `--queues` requires a command line argument, then priority classes should require a `--priority-classes` arg.